example-context/example-service/example-it/src/test/java/uk/gov/justice/services/example/cakeshop/it/CakeShopIT.java

```diff
@@ -452,32 +484,8 @@ private WebTarget sendTo(String url) {
                 ADD_RECIPE_MEDIA_TYPE);
     }
 
-
-    private InitialContext initialContext() throws NamingException {
-        Properties props = new Properties();
-        props.put(Context.INITIAL_CONTEXT_FACTORY, "org.jboss.naming.remote.client.InitialContextFactory");
-        props.put(Context.PROVIDER_URL, WILDFLY_REMOTING_URL);
-        props.put(Context.SECURITY_PRINCIPAL, JMS_USERNAME);
-        props.put(Context.SECURITY_CREDENTIALS, JMS_PASSWORD);
-        return new InitialContext(props);
-    }
-
-    private QueueReceiver queueReceiverOf(Context ctx, String queueName) throws NamingException, JMSException {
-        QueueConnectionFactory qconFactory = (QueueConnectionFactory) ctx.lookup(JMS_CONNECTION_FACTORY_JNDI);
-
-        QueueConnection qcon = qconFactory.createQueueConnection(JMS_USERNAME, JMS_PASSWORD);
-
-        QueueSession qsession = qcon.createQueueSession(false, Session.AUTO_ACKNOWLEDGE);
-        Queue queue = (Queue) ctx.lookup(queueName);
-        QueueReceiver qReceiver = qsession.createReceiver(queue);
-
-        qcon.start();
-
-        return qReceiver;
-    }
-
-    private void clear(QueueReceiver dlqReceiver) throws JMSException {
-        while (dlqReceiver.receiveNoWait() != null) {
+    private void clear(MessageConsumer msgConsumer) throws JMSException {
+        while (msgConsumer.receiveNoWait() != null) {
```

Comment:
Will this mechanism still work for topics? as it is individual subscriptions so you can't clear the topic from another subscribed message consumer. Might work for this test, but perhaps just thinking if this will be reusable?